zerver/lib/events.py

```diff
@@ -1820,12 +1864,9 @@ def do_events_register(
     else:
         event_types_set = None
 
-    # Fetch the realm object again to prefetch all the
-    # settings that will be used in 'fetch_initial_state_data'
-    # to avoid unnecessary DB queries.
-    # The settings include:
-    # * group settings which support anonymous groups
-    # * announcements streams
+    # Fetch the realm object again to prefetch
+    # `can_access_all_users_group` to avoid unnecessary
+    # DB queries.
     realm = get_realm_with_settings(realm_id=realm.id)
```

Comment:
Question: Does this actually save DB queries now? If it's just 1-2 queries to fetch when we need `can_access_all_users_group`, it might be better to just drop the 1 query here to refetch the realm object; which happens unconditionally, while the `can_access_all_users_group` call presumably only happens if you're a guest...

Reply:
Removed this code to fetch the realm again.